execution-engine/src/wasi/wasmi.rs

```diff
@@ -0,0 +1,1203 @@
+    const ERRNO: ValueType = ValueType::I32;
+    /// The representation type of the WASI `ExitCode` type.
+    const EXITCODE: ValueType = ValueType::I32;
+    /// The representation type of the WASI `FD` type.
+    const FD: ValueType = ValueType::I32;
+    /// The representation type of the WASI `FDFlags` type.
+    const FDFLAGS: ValueType = ValueType::I32;
+    /// The representation type of the WASI `FileDelta` type.
+    const FILEDELTA: ValueType = ValueType::I64;
+    /// The representation type of the WASI `FileSize` type.
+    const FILESIZE: ValueType = ValueType::I64;
+    /// The representation type of the WASI `FSTFlags` type.
+    const FSTFLAGS: ValueType = ValueType::I32;
+    /// The base pointer representation type of the WASI `IOVecArray` type, which
+    /// is passed as a pair of base address and length.
+    const IOVEC_ARRAY_BASE: ValueType = ValueType::I32;
+    /// The length representation type of the WASI `IOVecArray` type, which is
+    /// passed as a pair of base address and length.
+    const IOVEC_ARRAY_LENGTH: ValueType = ValueType::I32;
+    /// The representation type of the WASI `LookupFlags` type.
+    const LOOKUP_FLAGS: ValueType = ValueType::I32;
+    /// The representation type of the WASI `OFlags` type.
+    const OFLAGS: ValueType = ValueType::I32;
+    /// The representation type of the WASI `Rights` type.
+    const RIGHTS: ValueType = ValueType::I64;
+    /// The representation type of the WASI `SDFlags` type.
+    const SDFLAGS: ValueType = ValueType::I32;
+    /// The representation type of the WASI `SIFlags` type.
+    const SIFLAGS: ValueType = ValueType::I32;
+    /// The representation type of the WASI `RIFlags` type.
+    const RIFLAGS: ValueType = ValueType::I32;
+    /// The representation type of the WASI `Signal` type.
+    const SIGNAL: ValueType = ValueType::I32;
+    /// The representation type of the WASI `Size` type.
+    const SIZE: ValueType = ValueType::I32;
+    /// The representation type of the WASI `Timestamp` type.
+    const TIMESTAMP: ValueType = ValueType::I64;
+    /// The representation type of the WASI `Whence` type.
+    const WHENCE: ValueType = ValueType::I32;
+    /// The representation type of WASM `const` pointers (assuming `wasm32`).
+    const CONST_POINTER: ValueType = ValueType::I32;
+    /// The representation type of WASM pointers (assuming `wasm32`).
+    const POINTER: ValueType = ValueType::I32;
+    /// The representation type of WASM buffer length (assuming `wasm32`).
+    const SIZE_T: ValueType = ValueType::I32;
+
+    ////////////////////////////////////////////////////////////////////////////////
+    // Function well-formedness checks.
+    ////////////////////////////////////////////////////////////////////////////////
+
+    /// Checks the function signature, `signature`, has the correct type for the
+    /// host call coded by `index`.
+    pub(self) fn check_signature(index: WASIAPIName, signature: &Signature) -> bool {
```

Comment:
I think we can rename this to `WasiAPIName` which should be less jarring (Wasi is pronounced like a word, so need not be all capitals).